### ISSUE TITLE
[CI] Suite timeout in xpack.watcher.test.integration.SingleNodeTests

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#36065]
Build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/430/console

```
16:57:34   2> WARNING: Suite execution timed out: org.elasticsearch.xpack.watcher.test.integration.SingleNodeTests
16:57:34   2> ==== jstack at approximately timeout time ====
16:57:34   2> "elasticsearch[node_s0][management][T#2]" ID=247 TIMED_WAITING on org.elasticsearch.common.util.concurrent.EsExecutors$ExecutorScalingQueue@ddb4e9f
16:57:34   2> 	at sun.misc.Unsafe.park(Native Method)
16:57:34   2> 	- timed waiting on org.elasticsearch.common.util.concurrent.EsExecutors$ExecutorScalingQueue@ddb4e9f
16:57:34   2> 	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.awaitMatch(LinkedTransferQueue.java:734)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.xfer(LinkedTransferQueue.java:647)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.poll(LinkedTransferQueue.java:1277)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1073)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:57:34   2> 	at java.lang.Thread.run(Thread.java:748)
16:57:34   2> "elasticsearch[node_s0][watcher][T#2]" ID=246 WAITING on java.util.concurrent.CountDownLatch$Sync@3c9ce712
16:57:34   2> 	at sun.misc.Unsafe.park(Native Method)
16:57:34   2> 	- waiting on java.util.concurrent.CountDownLatch$Sync@3c9ce712
16:57:34   2> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
16:57:34   2> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
16:57:34   2> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
16:57:34   2> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
16:57:34   2> 	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
16:57:34   2> 	at org.elasticsearch.action.bulk.BulkRequestHandler.execute(BulkRequestHandler.java:85)
16:57:34   2> 	at org.elasticsearch.action.bulk.BulkProcessor.execute(BulkProcessor.java:372)
16:57:34   2> 	at org.elasticsearch.action.bulk.BulkProcessor.executeIfNeeded(BulkProcessor.java:363)
16:57:34   2> 	at org.elasticsearch.action.bulk.BulkProcessor.internalAdd(BulkProcessor.java:320)
16:57:34   2> 	- locked org.elasticsearch.action.bulk.BulkProcessor@42215896
16:57:34   2> 	at org.elasticsearch.action.bulk.BulkProcessor.add(BulkProcessor.java:303)
16:57:34   2> 	at org.elasticsearch.action.bulk.BulkProcessor.add(BulkProcessor.java:299)
16:57:34   2> 	at org.elasticsearch.action.bulk.BulkProcessor.add(BulkProcessor.java:285)
16:57:34   2> 	at org.elasticsearch.xpack.watcher.history.HistoryStore.put(HistoryStore.java:52)
16:57:34   2> 	at org.elasticsearch.xpack.watcher.execution.ExecutionService.execute(ExecutionService.java:318)
16:57:34   2> 	at org.elasticsearch.xpack.watcher.execution.ExecutionService.lambda$executeAsync$5(ExecutionService.java:398)
16:57:34   2> 	at org.elasticsearch.xpack.watcher.execution.ExecutionService$$Lambda$2830/108281761.run(Unknown Source)
16:57:34   2> 	at org.elasticsearch.xpack.watcher.execution.ExecutionService$WatchExecutionTask.run(ExecutionService.java:547)
16:57:34   2> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:660)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:57:34   2> 	at java.lang.Thread.run(Thread.java:748)
16:57:34   2> 	Locked synchronizers:
16:57:34   2> 	- java.util.concurrent.ThreadPoolExecutor$Worker@59d63772
16:57:34   2> "elasticsearch[node_s0][warmer][T#1]" ID=245 WAITING on org.elasticsearch.common.util.concurrent.EsExecutors$ExecutorScalingQueue@3a8cafa8
16:57:34   2> 	at sun.misc.Unsafe.park(Native Method)
16:57:34   2> 	- waiting on org.elasticsearch.common.util.concurrent.EsExecutors$ExecutorScalingQueue@3a8cafa8
16:57:34   2> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.awaitMatch(LinkedTransferQueue.java:737)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.xfer(LinkedTransferQueue.java:647)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.take(LinkedTransferQueue.java:1269)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:57:34   2> 	at java.lang.Thread.run(Thread.java:748)
16:57:34   2> "elasticsearch[node_s0][watcher][T#1]" ID=244 WAITING on java.util.concurrent.LinkedTransferQueue@6df4b1d0
16:57:34   2> 	at sun.misc.Unsafe.park(Native Method)
16:57:34   2> 	- waiting on java.util.concurrent.LinkedTransferQueue@6df4b1d0
16:57:34   2> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.awaitMatch(LinkedTransferQueue.java:737)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.xfer(LinkedTransferQueue.java:647)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.take(LinkedTransferQueue.java:1269)
16:57:34   2> 	at org.elasticsearch.common.util.concurrent.SizeBlockingQueue.take(SizeBlockingQueue.java:165)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:57:34   2> 	at java.lang.Thread.run(Thread.java:748)
16:57:34   2> "elasticsearch[node_s0][get][T#1]" ID=243 WAITING on java.util.concurrent.LinkedTransferQueue@25d9a110
16:57:34   2> 	at sun.misc.Unsafe.park(Native Method)
16:57:34   2> 	- waiting on java.util.concurrent.LinkedTransferQueue@25d9a110
16:57:34   2> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.awaitMatch(LinkedTransferQueue.java:737)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.xfer(LinkedTransferQueue.java:647)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.take(LinkedTransferQueue.java:1269)
16:57:34   2> 	at org.elasticsearch.common.util.concurrent.SizeBlockingQueue.take(SizeBlockingQueue.java:165)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:57:34   2> 	at java.lang.Thread.run(Thread.java:748)
16:57:34   2> "elasticsearch[node_s0][write][T#1]" ID=242 WAITING on java.util.concurrent.LinkedTransferQueue@7e2676ce
16:57:34   2> 	at sun.misc.Unsafe.park(Native Method)
16:57:34   2> 	- waiting on java.util.concurrent.LinkedTransferQueue@7e2676ce
16:57:34   2> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.awaitMatch(LinkedTransferQueue.java:737)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.xfer(LinkedTransferQueue.java:647)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.take(LinkedTransferQueue.java:1269)
16:57:34   2> 	at org.elasticsearch.common.util.concurrent.SizeBlockingQueue.take(SizeBlockingQueue.java:165)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
16:57:34   2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:57:34   2> 	at java.lang.Thread.run(Thread.java:748)
16:57:34   2> "elasticsearch[node_s0][search][T#2]" ID=241 WAITING on java.util.concurrent.LinkedTransferQueue@2188674d
16:57:34   2> 	at sun.misc.Unsafe.park(Native Method)
16:57:34   2> 	- waiting on java.util.concurrent.LinkedTransferQueue@2188674d
16:57:34   2> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
16:57:34   2> 	at java.util.concurrent.LinkedTransferQueue.awaitMatch(LinkedTransferQueue.java:737)
```

I cannot reproduce this locally, nevertheless:
```
./gradlew :x-pack:plugin:watcher:test \
  -Dtests.seed=3B8B6AE36FA884D2 \
  -Dtests.class=org.elasticsearch.xpack.watcher.test.integration.SingleNodeTests \
  -Dtests.security.manager=true \
  -Dtests.locale=en-US \
  -Dtests.timezone=Etc/UTC \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

**Comment by jdconrad** [comment]
This looks like a repeat of (https://github.com/elastic/elasticsearch/issues/36782).  Closing this in favor of the other issue.
